java/src/jmri/jmrit/automat/AbstractAutomaton.java

```diff
@@ -935,6 +936,7 @@ private boolean checkForConsistent(Turnout[] mTurnouts) {
 
     private DccThrottle throttle;
     private boolean failedThrottleRequest = false;
+    private ArrayList<ThrottleMemo> throttleListenerList = new ArrayList<ThrottleMemo>();
```

Comment:
Instead of retaining an ArrayList of ThrottleMemo, please use a [HashMap](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html) with DccThrottle as the key and ThrottleListener as the value.

Reply:
I did consider using a hashmap but decided against it because of the extra memory it would use. There would be performance gains when searching for the listener on release if the list is long but in practice the list will almost never be long enough to make a difference.

If you still think a hashmap is better I'm happy to do it that way. I don't know anything about the java hashmap implementation. My aversion is based on the Berkeley dbhash library which is lightening fast but a bit of a memory hog for small data sets.

Reply:
Definitely use the HashMap unless you can demonstrate that doing so will prevent the use of has severe negative impacts on its use. It is preferable to use a standard collection type when it will complete the task at hand vs loading a custom class definition to handle the same task. You have (possibly) traded some minor memory optimization in the collection for the overhead of yet another class.

Reply:
Haha, well I wouldn't want to be in your shoes when @bobjacobsen hears you've been dissing his classes! Anyway, Hashmaps it is.